ardupilot_gz_bringup/launch/robots/wildthumper.launch.py

```diff
@@ -0,0 +1,199 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Launch an wild thumper rover in Gazebo and Rviz.
+
+ros2 launch ardupilot_sitl sitl_dds_udp.launch.py
+transport:=udp4
+refs:=$(ros2 pkg prefix ardupilot_sitl)
+      /share/ardupilot_sitl/config/dds_xrce_profile.xml
+port:=2019
+synthetic_clock:=True
+wipe:=False
+model:=json
+speedup:=1
+slave:=0
+instance:=0
+defaults:=$(ros2 pkg prefix ardupilot_sitl)
+          /share/ardupilot_sitl/config/default_params/rover-skid.parm,
+          $(ros2 pkg prefix ardupilot_sitl)
+          /share/ardupilot_sitl/config/default_params/dds_udp.parm
+sim_address:=127.0.0.1
+master:=tcp:127.0.0.1:5760
+sitl:=127.0.0.1:5501
+"""
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.actions import IncludeLaunchDescription
+from launch.actions import RegisterEventHandler
+
+from launch.conditions import IfCondition
+
+from launch.event_handlers import OnProcessStart
+
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration
+from launch.substitutions import PathJoinSubstitution
+
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+
+
+def generate_launch_description():
+    """Generate a launch description for a wild thumper rover."""
+    pkg_ardupilot_sitl = get_package_share_directory("ardupilot_sitl")
+    pkg_ardupilot_sitl_models = get_package_share_directory("ardupilot_sitl_models")
+    pkg_project_bringup = get_package_share_directory("ardupilot_gz_bringup")
+
+    # Include component launch files.
+    sitl_dds = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [
+                PathJoinSubstitution(
+                    [
+                        FindPackageShare("ardupilot_sitl"),
+                        "launch",
+                        "sitl_dds_udp.launch.py",
+                    ]
+                ),
+            ]
+        ),
+        launch_arguments={
+            "transport": "udp4",
+            "refs": PathJoinSubstitution(
+                [
+                    FindPackageShare("ardupilot_sitl"),
+                    "config",
+                    "dds_xrce_profile.xml",
+                ]
+            ),
+            "port": "2019",
+            "command": "ardurover",
+            "synthetic_clock": "True",
+            "wipe": "False",
+            "model": "json",
+            "speedup": "1",
+            "slave": "0",
+            "instance": "0",
+            "defaults": os.path.join(
```

Comment:
In Pedro's work, we've been using pathlib for this operation. Would you consider switching?

Reply:
Can do. We have mixed approaches in various existing launch files, so should be consistent throughout.